### ISSUE TITLE
Implement setFullScreen() for Screens

### ISSUE THREAD

**Issue by samdeane** [Archaegeo/DualUniverseLuaIssues#48]
The new rendering API is funky. I'd love to use it to make consoles that you could use in an immersive way.

It would be great to have some sort of `setFullscreen` method on screen which aligned the camera directly in front of the screen and made it fill (or mostly fill) the user's view. It would also lock movement and free mouse and keyboard input.

**Comment by NQ-Ligo** [comment]
A camera control ? That's a sensitive thing.

This should be first a non-Lua feature, before being able to add it on Lua. But could be something really nice (it will require the loacl Lua authorization for that anyway).

I like the idea anyway

**Comment by samdeane** [comment]
Yes, I think you’d have to explicitly interact with the screen in question; much the way you do with things like the market consoles. 

In effect that’s the experience I’m looking to be able to reproduce - a console of some sort that you can focus on so that you don’t have to suffer your viewpoint wavering around.

I guess it could be completely non-lua, and just a checkbox on the screen “Can fullscreen” - which would unlock some game UI.

As a bonus: being “fullscreened” on a screen would allow the game to feed it keypress events, which would allow us to implement proper text input on a screen.
